src/index/base.cpp

```diff
@@ -138,7 +134,7 @@ void BaseIndex::ThreadSync()
         int64_t last_locator_write_time = 0;
         while (true) {
             if (m_interrupt) {
-                m_best_block_index = pindex;
```

Comment:
Haven't tried this, but what happens if `m_best_block_index` is private?

Reply:
Pulling some of these into #24150 (as it needs to be rebased anyway now).

Reply:
I'm not sure I can follow: `m_best_block_index` is already private?